File: packages/api/src/@types/PoolState.ts
import type PoolInfo from './PoolInfo';

type PoolState = {
  p2_singleton_puzzle_hash: string;
  points_found_since_start: number;
  points_found_24h: [number, number][];
  points_acknowledged_since_start: number;
  points_acknowledged_24h: [number, number][];
  current_points: number;
  current_difficulty: number;
  plotCount: number;
  pool_errors_24h: {
    current_difficulty: number;
    error_code: number;
    error_message: string;
  }[];
  pool_info: PoolInfo;
  pool_config: {
    authentication_key_info_signature: string;
    authentication_public_key: string;
    authentication_public_key_timestamp: number;
    owner_public_key: string;
    pool_puzzle_hash: string;
    pool_url: string;
    launcher_id: string;
    target: string;
    target_signature: string;
    payout_instructions: string;
    target_puzzle_hash: string;
  };
};

export default PoolState;

File: chia/pools/pool_state.py
from __future__ import annotations

from typing import Any, Dict, List, Optional, Tuple
from typing_extensions import TypedDict

from chia.pools.pool_config import PoolWalletConfig
from chia.util.ints import uint8, uint64


class PoolState(TypedDict):
    p2_singleton_puzzle_hash: str
    points_found_since_start: uint64
    points_found_24h: List[Tuple[float, uint64]]
    points_acknowledged_since_start: uint64
    points_acknowledged_24h: List[Tuple[float, uint64]]
    next_farmer_update: float
    next_pool_info_update: float
    current_points: uint64
    current_difficulty: Optional[uint64]
    pool_errors_24h: List[Dict[str, Any]]
    authentication_token_timeout: Optional[uint8]
    pool_config: PoolWalletConfig
    plot_count: int
